### ISSUE TITLE
H calculate

### ISSUE THREAD

**Issue by yiqingyang2012** [JeremyCCHsu/Gumbel-Softmax-VAE-in-tensorflow#1]
@JeremyCCHsu 
when calc H, you use sampled q(y)
                loss['H(y)'] = tf.reduce_mean(                
                    tf.reduce_sum(
                        tf.mul(
                            unlabel['y_sample'],
                            tf.log(unlabel['y_sample'] + EPS) - tf.log(y_prior)),
                        -1))

why don't you use the y_pred,like below:
                loss['H(y)'] = tf.reduce_mean(                
                    tf.reduce_sum(
                        tf.mul(
                            unlabel['y_pred'],
                            tf.log(unlabel['y_pred'] + EPS) - tf.log(y_prior)),
                        -1))

**Comment by JeremyCCHsu** [comment]
@yiqingyang2012 You're right, it should be calculated using the inferred `y` as it should be Categorically distributed.
